azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/implementation/FunctionDeploymentSlotImpl.java

```diff
@@ -0,0 +1,376 @@
+    }
+
+    @Override
+    public Completable stopAsync() {
+        return manager().inner().webApps().stopSlotAsync(resourceGroupName(), this.parent().name(), name())
+                .flatMap(new Func1<Void, Observable<?>>() {
+                    @Override
+                    public Observable<?> call(Void aVoid) {
+                        return refreshAsync();
+                    }
+                }).toCompletable();
+    }
+
+    @Override
+    public void restart() {
+        restartAsync().toObservable().toBlocking().subscribe();
+    }
+
+    @Override
+    public Completable restartAsync() {
+        return manager().inner().webApps().restartSlotAsync(resourceGroupName(), this.parent().name(), name())
+                .flatMap(new Func1<Void, Observable<?>>() {
+                    @Override
+                    public Observable<?> call(Void aVoid) {
+                        return refreshAsync();
+                    }
+                }).toCompletable();
+    }
+
+    @Override
+    public FunctionDeploymentSlotImpl withBrandNewConfiguration() {
+        this.siteConfig = null;
+        return this;
+    }
+
+    @Override
+    public FunctionDeploymentSlotImpl withConfigurationFromParent() {
+        return withConfigurationFromWebApp(this.parent());
+    }
+
+    @Override
+    public FunctionDeploymentSlotImpl withConfigurationFromWebApp(FunctionApp app) {
+        copyConfigurations(((WebAppBaseImpl) app).siteConfig, app.appSettings().values(), app.connectionStrings().values());
+        return this;
+    }
+
+    @Override
+    public FunctionDeploymentSlotImpl withConfigurationFromDeploymentSlot(FunctionDeploymentSlot slot) {
+        copyConfigurations(((WebAppBaseImpl) slot).siteConfig, slot.appSettings().values(), slot.connectionStrings().values());
+        return this;
+    }
+
+    private void copyConfigurations(SiteConfigResourceInner configInner, Collection<AppSetting> appSettings, Collection<ConnectionString> connectionStrings) {
+        this.siteConfig = configInner;
+        // app settings
+        for (AppSetting appSetting : appSettings) {
+            if (appSetting.sticky()) {
+                withStickyAppSetting(appSetting.key(), appSetting.value());
+            } else {
+                withAppSetting(appSetting.key(), appSetting.value());
+            }
+        }
+        // connection strings
+        for (ConnectionString connectionString : connectionStrings) {
+            if (connectionString.sticky()) {
+                withStickyConnectionString(connectionString.name(), connectionString.value(), connectionString.type());
+            } else {
+                withConnectionString(connectionString.name(), connectionString.value(), connectionString.type());
+            }
+        }
+    }
+
+    @Override
+    public FunctionAppImpl parent() {
+        return this.parent;
+    }
+
+    @Override
+    Observable<SiteInner> createOrUpdateInner(SiteInner site) {
```

Comment:
Ideally these can be shared between `DepoymentSlotImpl` and `FunctionDeploymentSlotImpl`, just like `WebAppImpl` and `FunctionAppImpl` share these methods in `AppServceBaseImpl`.